### ISSUE TITLE
Header name must be a valid HTTP Token

### ISSUE THREAD

**Issue by alexey-sh** [request/request#2732]
### Simplest Example to Reproduce

```js

var request = require('request');

var options = {
    url: '----',
    headers: {
        ':authority': 'www.site.com',
        ':method': 'GET',
        ':scheme': 'https',
        'referer': 'https://site.com',
        'User-Agent': 'Mozilla/5.0 (Macintosh; Intel Mac OS X 10_12_6) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/60.0.3112.78 Safari/537.36'
    }
};

function callback(error, response, body) {
    if (!error && response.statusCode === 200) {
        console.log(body);
    }
    if (error) {
        console.log(error);
    }
}

request(options, callback);

```

### Your Environment

| software         | version
| ---------------- | -------
| request          | 2.81.0
| node             | v5.10.1
| npm              | 5.3.0
| Operating System | mac os 10.12.6

**Comment by tommedema** [comment]
Got the same issue, but would like to set header to `:authority:` instead of `:authority`. Seems like this is in node-core, is that right? I can't seem to find the error line in request itself.

**Comment by alexey-sh** [comment]
@tommedema I think so too. I decided to remove all colons from the headers. I would like to recommend you the same approach.

**Comment by AshleyMcVeigh** [comment]
Same issue, I'm trying to send a request with a :authority header, is there a workaround?

**Comment by tommedema** [comment]
@alexey-sh that doesn't work for me because I don't control the server side, only the client side (request)

**Comment by alexey-sh** [comment]
@tommedema did you tried without colons? I didn't have control too.

**Comment by Freyert** [comment]
Your API has to be fixed:

```
3.2.  Header Fields

   Each header field consists of a case-insensitive field name followed
   by a colon (":"), optional leading whitespace, the field value, and
   optional trailing whitespace.
```

**Comment by tommedema** [comment]
@Freyert I don't agree that this is a solution because it is not guaranteed that this API is under our control. At least this should be configurable. Many of us will want to be able to request APIs that do not adhere perfectly to IETF standards.

**Comment by kerkero** [comment]
It's certainly not a solution, but as a workaround I wrapped [node-libcurl](https://github.com/JCMais/node-libcurl) which uses cUrl under the hood and allows headers which start with colons.

**Comment by G-Rath** [comment]
@tommedema I'd say the bare minimal & default should be to follow the RFC first, user prefer second.

> Many of us will want to be able to request APIs that do not adhere perfectly to IETF standards.

Right now I want to make a request to an API that **does** adhere perfectly to IETF standards, but can't b/c of this issue, which hasn't been resolved, despite the fact that it means `request` currently doesn't conform to IETF standards.

I'm totally on your side with this one about that it should be configurable, but as I said at the start of this comment; the bare-minimal should to follow the IETF standard, as that's why its there in the first place.

**Comment by tommedema** [comment]
Sure, I agree.

**Comment by akshayjai1** [comment]
for any newbie 👶 understands
```
GET http://localhost:3099/subscribers
GET http://localhost:3099/subscribers
```
>will throw error
**While, following will work well**
```
GET http://localhost:3099/subscribers

GET http://localhost:3099/subscribers
```
